src/Analyzers/VerifyShouldBeUsedOnlyForOverridableMembersAnalyzer.cs

```diff
@@ -0,0 +1,195 @@
+                return;
+            }
+
+            // If not overridable, fall through to report diagnostic
+        }
+        else
+        {
+            if (IsAllowedMockMember(mockedMemberSymbol, knownSymbols))
+            {
+                return;
+            }
+        }
+
+        // 6. If we reach here, the member is neither overridable nor allowed by Moq
+        //    So we report the diagnostic.
+        //
+        // NOTE: The location is on the invocationOperation, which is fairly broad
+        // For VerifySet, try to report the diagnostic on the property being set for precise span
+        Location diagnosticLocation = invocationOperation.Syntax.GetLocation();
+
+        Diagnostic diagnostic = DiagnosticExtensions.CreateDiagnostic(invocationOperation.Syntax, Rule, diagnosticLocation);
+        context.ReportDiagnostic(diagnostic);
+    }
+
+    /// <summary>
+    /// Determines whether a member can be mocked.
+    /// </summary>
+    /// <param name="mockedMemberSymbol">The mocked member symbol.</param>
+    /// <param name="knownSymbols">The known symbols.</param>
+    /// <returns>
+    /// Returns <see langword="true"/> when the diagnostic should not be triggered; otherwise <see langword="false" />.
+    /// </returns>
+    private static bool IsAllowedMockMember(ISymbol mockedMemberSymbol, MoqKnownSymbols knownSymbols)
+    {
+        switch (mockedMemberSymbol)
+        {
+            case IPropertySymbol propertySymbol:
+                return propertySymbol.IsOverridable() || propertySymbol.IsTaskOrValueResultProperty(knownSymbols);
+
+            case IMethodSymbol methodSymbol:
+                return methodSymbol.IsOverridable();
+
+            default:
+                // If it's not a property or method, it can't be mocked. This includes fields and events.
+                return false;
+        }
+    }
+
+    /// <summary>
+    /// Attempts to resolve the symbol representing the member (property or method)
+    /// being referenced in the Verify(...) or VerifySet(...) call. Returns null if it cannot be determined.
+    /// </summary>
+    private static ISymbol? TryGetMockedMemberSymbol(IInvocationOperation moqVerifyInvocation, MoqKnownSymbols knownSymbols)
+    {
+#pragma warning disable S125 // Sections of code should not be commented out
+        // Usually the first argument to a Moq Verify(...) is a lambda expression like x => x.Property
+        // or x => x.Method(...). For VerifySet, it's an Action<T> lambda: x => { x.Property = ...; }
+#pragma warning restore S125 // Sections of code should not be commented out
+        if (moqVerifyInvocation.Arguments.Length == 0)
+        {
+            return null;
+        }
+
+        // For VerifySet, the lambda may be in argument 1 (second argument)
+        int lambdaArgIndex = 0;
+        if (moqVerifyInvocation.TargetMethod.IsInstanceOf(knownSymbols.Mock1VerifySet) && moqVerifyInvocation.Arguments.Length > 1)
+        {
+            lambdaArgIndex = 1;
```

Comment:
This line is not covered. Evaluate if the conditional check on about L150 is valid. If so, write a test that covers this scenario. If not, refactor.